pycqed/analysis_v2/tomography_V2.py

```diff
@@ -0,0 +1,553 @@
+            return rhos
+
+
+
+    def calculate_LI_coefficient_matrix(self, measurement_operator, do_in_pauli=False):
+        """
+        Calculates the coefficient matrix used when inversing the linear system of equations needed to find rho
+        Requires a calibrated measurement operator
+        if do_in_pauli is true this will presume measurement operator is given in the  basis.
+        """
+        coefficient_matrix = np.zeros((len(self.rotation_matrixes) ** self.n_qubits, 4 ** self.n_qubits), dtype=complex)
+        Ms = self.get_measurement_vector(measurement_operator, do_in_pauli=do_in_pauli)
+        for i in range(len(Ms)):
+            coefficient_matrix[i,:] =  np.ravel(Ms[i].full())
+        return coefficient_matrix
+
+    def get_measurement_vector(self, measurement_operator, do_in_pauli=False):
+        """
+        Returns a list of rotated measurement operator based on an initial measurement operator which should be obtained from the calibration
+        """
+        n_rotations = len(self.rotation_matrixes) ** self.n_qubits
+        measurement_vector = []
+        for i in range(n_rotations):
+            R = self.rotation_vector[i]
+            if do_in_pauli:
+                M = self.trans_comp_to_pauli(R.dag() * measurement_operator * R)
+            else:
+                M = R.dag() * measurement_operator * R
+            measurement_vector.append(M)
+        return measurement_vector
+
+    def calibrate_measurement_operator(self, cal_meas, calibration_points=None, TE_correction_matrix=None, transform_to_pauli=False):
+        """
+        Calibrates the measurement operator in any basis. Assumes cal_meas are the eigenvalues of the calibration_points
+        defaults the calibration_points to be the computational basis projectors. 
+        If TE_correction_matrix is set, this alters the computational basis states
+        """
+        if calibration_points is None:
+            calibration_points = comp_projectors
+        M = sum([cal_meas[i] * calibration_points[i] for i in range(len(cal_meas))])
+        return M if not transform_to_pauli else self.trans_comp_to_pauli(M)
+
+
+    def calibrate_bin_operators(self, calibration_counts, calibration_points=None, normalize=False):
+        M_bins = []
+        #calculate bin probabilities normalized horizontally
+        if normalize:
+            cal_probs = (calibration_counts / np.sum(calibration_counts, axis = 1, dtype=float)[:,np.newaxis])
+        else:
+            cal_probs = np.array(calibration_counts)#(calibration_counts / np.sum(calibration_counts, axis = 1, dtype=float)[:,np.newaxis])
+        for probs in cal_probs.T:
+            #calibrate the measurement operator for each bin in the same way as done with average tomo. 
+            M_bins.append(self.calibrate_measurement_operator(probs, calibration_points))
+        return M_bins
+
+    
+
+
+
+#########################################################################
+#
+# HELPERS
+#
+
+
```

Comment:
The helper methods look like they can (in general) be functions (not part of the class, not having the self keyword). 
The advantage of moving them outside of the class is that they can be used in other applications and it is clear to the user (readability) that these are pure functions (that do not require knowledge of the current state of the object).